core/htmlparser/filter.js

```diff
@@ -402,6 +402,144 @@
 } )();
 
 /**
+ * Abstract class describing the definition of {@link CKEDITOR.htmlParser.filter} rules.
+ *
+ * Definition object represents rules as a set of properties with callback functions
+ * to be applied for transforming and filtering content upon data processing.
+ *
  * @class CKEDITOR.htmlParser.filterRulesDefinition
  * @abstract
  */
+
+/**
+ * @property {Array<String,RegExp>[]} elementNames An array of rules for element names transformation.
+ * Every rule match will be replaced by the given string.
+ *
+ * Examples:
+ *
+ * ```javascript
+ * elementNames: [
+ * 		[ /^div$/, 'p' ], // Converts 'div' into 'p'.
+ * 		[ /^cke:?/, '' ] // Removes 'cke:' prefixes.
+ * ]
+ * ```
+ *
+ */
+
+/**
+ * @property {Array<String,RegExp>[]} attributeNames An array of rules for attribute names transformation.
+ * Every matching string from the first item will be converted into a second.
+ *
+ * Examples:
+ *
+ * ```javascript
+ * attributeNames: [
+ * 		[ 'data-foo', 'data-bar' ],
+ * 		// Converts string in attribute name from 'data-foo' into 'data-bar'
+ * 		// Note that attribute 'data-foo-baz' will be converted into 'data-bar-baz'.
+ *
+ * 		[ /^data-custom$/, 'data-cke' ]
+ * 		// Converts attribute 'data-custom' into 'data-cke'.
+ * ]
+ * ```
+ *
+ */
+
+/**
+ * @property {Object.<String, Function>} elements An object containing pairs of element selectors
+ * and functions used upon element filtering and transformation.
+ *
+ * A selector can be either element name or one of following: '^', '$'.
+ *
+ * '^' and '$' is to be applied on every filtered element. The first is applied before element specific filter,
+ * and second is applied after element specific filter.
+ *
+ * Function can contain return statement:
+ * - If `false` is returned the element is removed.
+ * - If an other element is returned it overwrites the original element.
```

Comment:
Actually, returning element always overwrites the original element, although new type element won't be processed anymore: https://github.com/ckeditor/ckeditor-dev/blob/master/core/htmlparser/filter.js#L347
What do you think about:

> Additionaly function can contain return statement:                                                                                                                                                                                    
> - `false` will remove whole element from the node tree.                                                                                                                                                                               
> - {@link CKEDITOR.dom.element element} will overwrite the original element.